examples/solvers/chambolle_pock_tomography_tgv.py

```diff
@@ -1,116 +1,136 @@
-"""Total generalized variation tomography using the Chambolle-Pock solver.
+"""Total Generalized Variation tomography using the Chambolle-Pock solver.
 
 Solves the optimization problem
 
-    min_x ||A(x) - g||_2^2 + TGV_2(x)
+    min_x ||Ax - d||_2^2 + alpha * TGV_2(x)
 
-Where ``A`` is a parallel beam forward projector and ``g`` is given noisy data.
+Where ``A`` is a parallel beam forward projector and ``d`` is given noisy data.
 TGV_2 is the second order total generalized variation of ``x``, defined as
 
-    TGV_2(x) = min_y lam_1 ||grad(x) - y||_1 + lam_2 ||eps(y)||_1
+    TGV_2(x) = min_y ||Gx - y||_1 + beta ||Ey||_1
 
-where ``grad`` is the (vectorial) spatial gradient and ``eps`` is the matrix
-valued spatial derivative of the vector valued function ``y``.
+where ``G`` is the spatial gradient operating on the scalar-field ``x`` and
+``E`` is the symmetrized gradient operating on the vector-field ``y``.
+Both one-norms ||.||_1 take the one-norm globally and the two-norm locally,
+i.e. ||y||_1 := sum_i sqrt(sum_j y_i(j)^2) where y_i(j) is the j-th value
+of the vector y_i at location i.
 
 The problem is rewritten as
 
-    min_{x, y} ||A(x) - g||_2^2 +  lam_1 ||grad(x) - y||_1 + lam_2 ||eps(y)||_1
+    min_{x, y} ||Ax - d||_2^2 + alpha ||Gx - y||_1 + alpha * beta ||Ey||_1
 
 which can then be solved with the Chambolle-Pock method.
+
+References
+----------
+[1] K. Bredies and M. Holler. A TGV-based framework for variational image
+decompression, zooming and reconstruction. Part II: Numerics. SIAM Journal
+on Imaging Sciences, 8(4):2851-2886, 2015.
 """
 
 import numpy as np
 import odl
 
-
 # --- Set up the forward operator (ray transform) --- #
 
-
 # Reconstruction space: discretized functions on the rectangle
-# [-20, 20]^2 with 300 samples per dimension.
-reco_space = odl.uniform_discr(
-    min_pt=[-20, -20], max_pt=[20, 20], shape=[300, 300], dtype='float32')
+# [-20, 20]^2 with 100 samples per dimension.
+U = odl.uniform_discr(
+    min_pt=[-20, -20], max_pt=[20, 20], shape=[100, 100], dtype='float32')
 
 # Make a parallel beam geometry with flat detector
-geometry = odl.tomo.parallel_beam_geometry(reco_space)
+geometry = odl.tomo.parallel_beam_geometry(U)
 
 # Create the forward operator
-ray_trafo = odl.tomo.RayTransform(reco_space, geometry)
-
+A = odl.tomo.RayTransform(U, geometry)
 
 # --- Generate artificial data --- #
 
-
 # Create phantom
-phantom = odl.phantom.shepp_logan(reco_space, modified=True)
+def phantom_func(pts):
+    x, y = pts
+    sign = ((x < -10) | (x > 10) | (y < -10) | (y > 10)) * 3 - 2
+    return (x / 20.0) * sign
+
+phantom = U.element(phantom_func)
+phantom.show(title='Phantom')
 
 # Create sinogram of forward projected phantom with noise
-data = ray_trafo(phantom)
-data += odl.phantom.white_noise(ray_trafo.range) * np.mean(data) * 0.1
+data = A(phantom)
+data += odl.phantom.white_noise(A.range) * np.mean(data) * 0.1
```

Comment:
This should be `np.mean(np.abs(data))`, since otherwise you get 0 noise. Noticed this while running the examples